apps/src/clientApi.js

```diff
@@ -96,12 +97,25 @@ class CollectionsApi {
   }
 
   _withBeforeFirstWriteHook(fn) {
-    if (this._beforeFirstWriteHook) {
-      this._beforeFirstWriteHook(err => {
-        // continuing regardless of error status from hook...
-        fn();
+    if (this._shouldCallBeforeFirstWriteHook && this._beforeFirstWriteHook) {
+      this._beforeFirstWriteHook((err, success) => {
+        if (err) {
+          firehoseClient.putRecord(
+            {
+              study: 'weblab_loading_investigation',
+              study_group: 'empty_manifest',
+              event: 'error_uploading_starter_files',
+              project_id: this.getProjectId()
+            },
+            {includeUserId: true}
+          );
+          this._shouldCallBeforeFirstWriteHook = true;
+          this._errorAction();
+        } else if (success) {
+          this._shouldCallBeforeFirstWriteHook = false;
+          fn();
+        }
       });
-      this._beforeFirstWriteHook = null;
```

Comment:
Do we still need to be setting this to null in success case?

Reply:
It's not technically needed, but is some good cleanup to do.